model/src/main/scala/uk/gov/ons/addressIndex/model/server/response/AddressResponse.scala

```diff
@@ -189,9 +198,22 @@ object AddressResponseAddress {
       northing <- Try(other.northing.toInt)
     } yield AddressResponseGeo(latitude, longitude, easting, northing)
 
+    val saoHyphen = if ((other.saoStartNumber != "" || other.saoStartSuffix != "")
```

Comment:
Its quite unclear whats going on here, maybe separate your booleans out to descriptive vars to get a better grasp of what's going on.